Foundation/NSCalendar.swift

```diff
@@ -1425,7 +1425,11 @@ open class NSDateComponents : NSObject, NSCopying, NSSecureCoding {
     }
     
     open func copy(with zone: NSZone? = nil) -> Any {
-        NSUnimplemented()
+        let copy = NSDateComponents()
+        copy._values = self._values
+        copy._calendar = self._calendar
+        copy._timeZone = self._timeZone
+        return copy
```

Comment:
@phausler Thank you for looking into this.

Wrote a code snippet to subclass NSDateComponents:
```
class Test: NSDateComponents { 
 }
 
func test_subclass() { 
        let obj = Test()
        obj.year = 1987
        obj.month = 3
        obj.day = 17
        obj.hour = 14
        obj.minute = 20
        obj.second = 0
        let newObj = obj.copy(with: nil) as! NSDateComponents
        print(newObj.year)
        print(newObj.month)
        print(newObj.day)
    }
```
Executing the above prints the year month and day as expected. 

Are there shortcomings for subclassers that I am missing?

Thanks.